tests/app/main/test_frameworks.py

```diff
@@ -1640,6 +1629,137 @@ def test_make_or_edit_supplier_declaration_shows_correct_page(self, data_api_cli
+class TestFrameworksDashboardConfidenceBannerOnPage(BaseApplicationTest):
+    """Tests for the confidence banner on the declaration page."""
+
+    expected = (
+        'Your application will be submitted at 5pm BST, 23 June 2016. <br> '
+        'You can edit your declaration and services at any time before the deadline.'
+    )
+
+    def test_confidence_banner_on_page(self, data_api_client, _):
+        """Test confidence banner appears on page happy path."""
+        data_api_client.get_framework.return_value = self.framework(status='open')
+        data_api_client.find_draft_services.return_value = {
+            "services": [
+                {'serviceName': 'A service', 'status': 'submitted', 'lotSlug': 'foo'}
+            ]
+        }
+        data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(status='complete')
+
+        with self.app.test_client():
+            self.login()
+            res = self.client.get("/suppliers/frameworks/g-cloud-8")
+        assert res.status_code == 200
+        assert self.expected in str(res.data)
+
+    def test_confidence_banner_not_on_page(self, data_api_client, _):
+        """Change value and assertt that confidence banner is not displayed."""
+        data_api_client.get_framework.return_value = self.framework(status='open')
+        data_api_client.find_draft_services.return_value = {
+            "services": [
+                {'serviceName': 'A service', 'status': 'not-submitted', 'lotSlug': 'foo'}
+            ]
+        }
+        data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(status='complete')
+
+        with self.app.test_client():
+            self.login()
+            res = self.client.get("/suppliers/frameworks/g-cloud-8")
+        assert res.status_code == 200
+        assert self.expected not in str(res.data)
+
+
+@mock.patch('dmutils.s3.S3')
+@mock.patch('app.main.views.frameworks.flash', autospec=True)
+@mock.patch('app.main.views.frameworks.data_api_client', autospec=True)
+class TestFrameworksDashboardConfidenceBannerFlashes(BaseApplicationTest):
+    """Tests for the confidence banner on the declaration page."""
+
+    def test_confidence_banner_flash(self, data_api_client, flash_mock, _):
+        """Test confidence banner happy path."""
+        data_api_client.get_framework.return_value = self.framework(status='open')
+        data_api_client.find_draft_services.return_value = {
+            "services": [
+                {'serviceName': 'A service', 'status': 'submitted', 'lotSlug': 'foo'}
+            ]
+        }
+        data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(status='complete')
+
+        with self.app.test_client():
+            self.login()
+            resp = self.client.get("/suppliers/frameworks/g-cloud-8")
+
+        assert resp.status_code == 200
+        # Assert mock.
+        flash_mock.assert_called_once_with('submission_confidence', 'success')
```

Comment:
The idea of this test is to monitor the boundary contract between code and template. If the view doesn't flash the appropriate key then there will be a problem in the template ie the contract between the view and template has changed.